pkg/apis/flow/v1alpha1/synchronizer_lifecycle.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+// GetGroupVersionKind implements kmeta.OwnerRefable.
+func (s *Synchronizer) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("Synchronizer")
+}
+
+// SynchronizerCondSet is the group of possible conditions
+var SynchronizerCondSet = apis.NewLivingConditionSet(
+	ConditionSinkProvided,
+	ConditionDeployed,
+)
+
+// InitializeConditions sets relevant unset conditions to Unknown state.
+func (s *SynchronizerStatus) InitializeConditions() {
+	SynchronizerCondSet.Manage(s).InitializeConditions()
+}
```

Comment:
```suggestion

```
The Ready condition should never be set by us. Knative will set it automatically based on the other conditions in the ConditionSet.

Reply:
Yes, I know that but how Ready condition is related to condition set initialization? Please elaborate on that.

Reply:
I think Jeff copy pasted a comment I left on his own PR but copied the wrong one.

The true rationale is that, if your type implements KRShaped, conditions are always initialized automatically for you in the code generated by codegen.

Reply:
Ah, I see, thanks